src/main/java/com/metamx/http/client/HttpClientConfig.java

```diff
@@ -24,6 +24,35 @@
  */
 public class HttpClientConfig
 {
+  public enum CompressionCodec
+  {
+    IDENTITY {
+      @Override
+      public String getEncodingString()
+      {
+        return "identity";
+      }
+    },
+    GZIP {
+      @Override
+      public String getEncodingString()
+      {
+        return "gzip";
+      }
+    },
+    DEFLATE {
+      @Override
+      public String getEncodingString()
+      {
+        return "deflate";
+      }
+    };
+
+    public abstract String getEncodingString();
```

Comment:
I did this because I'm anticipating that in the future we might want to add LZ4, which would manifest as an enum value `LZ4` but encoding string "x-lz4"